src/events/common/metrics-context.ts

```diff
@@ -14,7 +14,11 @@ export function metricsContext(
     } = {}
 ) {
     function publishMetrics() {
-        metrics.instance.publishStoredMetrics()
+        // prevent logging an error inside the cloudwatch logs
+        const storedMetrics = (metrics.instance as unknown as { storedMetrics?: object }).storedMetrics
+        if (Object.keys(storedMetrics ?? {}).length > 0) {
```

Comment:
```suggestion
        if (storedMetrics === undefined || Object.keys(storedMetrics).length > 0) {
```
This approach would still publish metrics (with a warning) in case the internals of Powertools change. What do you think?